lib/spring/application.rb

```diff
@@ -67,6 +67,11 @@ def initialized?
     def start_watcher
       @watcher = Spring.watcher
       @watcher.on_stale { state! :watcher_stale }
+
+      @watcher.on_debug do |message|
+        spring_env.log "[watcher:#{app_env}] #{message}"
+      end if @watcher.respond_to?(:on_debug)
```

Comment:
Only the poller watcher implements it. Calling it on the listen watcher would bomb.